pkg/iac/scanners/terraform/parser/evaluator.go

```diff
@@ -310,9 +332,12 @@ func (e *evaluator) expandBlockForEaches(blocks terraform.Blocks, isDynamic bool
 			// instances are identified by a map key (or set member) from the value provided to for_each
 			idx, err := convert.Convert(key, cty.String)
 			if err != nil {
-				e.debug.Log(
-					`Invalid "for-each" argument: map key (or set value) is not a string, but %s`,
-					key.Type().FriendlyName(),
+				e.logger.Error(
+					`Failed to expand block. Invalid "for-each" argument: map key (or set value) is not a string`,
```

Comment:
@nikpivkin Hey there. It seems after this change, a couple of previously debug log now showing as error log. do you have any suggestion to keep the logs output roughly as they were so it doesn't confuse my IaC modules' users than needed? Thanks a lot!

Reply:
@simar7 Maybe we should lower the message level to `debug`?

Reply:
@nikpivkin I agree, could you open a PR to address that?

Reply:
Thank you both for the quick response and action!